distributedID/blockchain-explorer
# Removing the package-lock.json files.

**Ref:**  #90 

**Description:**

`package-lock.json` makes sure that the packages (`npm packages`) when being installed via the `npm install` command is so in a specific sequence of package version. This makes sure that the remaining packages, if dependent on other packages, are not installed `twice` or break due to the unavailability of the packages.

While this is useful in most cases, in our `explorer` it looks like it is not an absolute requirement. By deleting these files, we can reduce the `repo` size by `100s of MBs`. Which in turn will help us in creating the containers locally with speed.

Also, due to its huge size, `cloning` the repo becomes a problem.

**Blocking:** Nah

**Testing:**

For the `blockchain-explorer/client` directory:

- Delete the `node_modules` directory.
- Run the `npm install`.
- Build the `react` code with `REACT_APP_CLIENT=Provider npm run build`.

Now, move to the root directory of the repo, `blockchain-explorer`:

- Delete the `node_modules` directory.
- Run the `npm install`.

Make sure that the `bcmanager` network is up and running. Once that is done, use
`REACT_APP_CLIENT=Provider npm start`
command from the `blockchain-explorer` directory and access the `explorer` on `localhost:11000` address. 

Everything should be working fine.
